src/cc/api/BPF.cc

```diff
@@ -650,6 +652,25 @@ BPFStackTable BPF::get_stack_table(const std::string& name, bool use_debug_file,
   return BPFStackTable({}, use_debug_file, check_debug_file_crc);
 }
 
+BPFStackBuildIdTable BPF::get_stackbuildid_table(const std::string &name, bool use_debug_file,
+                                                 bool check_debug_file_crc) {
+  TableStorage::iterator it;
+
+  if (bsymcache_ == NULL) {
+    bsymcache_ = bcc_buildsymcache_new();
+  }
+
+  if (bpf_module_->table_storage().Find(Path({bpf_module_->id(), name}), it))
+    return BPFStackBuildIdTable(it->second, use_debug_file, check_debug_file_crc, bsymcache_);
+  return BPFStackBuildIdTable({}, use_debug_file, check_debug_file_crc, bsymcache_);
+}
+
+bool BPF::add_module(std::string module)
+{
+  return bcc_buildsymcache_add_module(bsymcache_, module.c_str()) != 0 ?
+    false : true;
```

Comment:
Currently, C++ application needs to do get_stackbuildid_table first before calling add_module. Do you think in BPF::add_module() call, we should also call
```
  if (bsymcache_ == NULL) {
    bsymcache_ = bcc_buildsymcache_new();
  }
```
so that user can add modules before calling `get_stackbuildid_table`?

I agree with your current implementation that a bsymcache is built only if it may be used.

Reply:
Point taken. Made the necessary change.